--- options.mk.orig	2015-07-20 22:45:00.000000000 -0700
+++ options.mk	2015-07-21 09:11:37.000000000 -0700
@@ -1,9 +1,13 @@
 # $NetBSD: options.mk,v 1.8 2015/01/21 13:45:18 pho Exp $
 
 PKG_OPTIONS_VAR=	PKG_OPTIONS.dbus
-PKG_SUPPORTED_OPTIONS+=	debug x11
+PKG_SUPPORTED_OPTIONS+=	debug x11 enable_in_rcvar
 PKG_SUGGESTED_OPTIONS=	x11
 
+.if ${OPSYS} == "FreeBSD" || ${OPSYS} == "DragonFly"
+PKG_SUGGESTED_OPTIONS+= enable_in_rcvar	
+.endif
+
 .if (${OPSYS} == "NetBSD"  ||	\
      ${OPSYS} == "FreeBSD" ||	\
      ${OPSYS} == "OpenBSD" ||	\
@@ -22,31 +26,41 @@
 .include "../../mk/bsd.options.mk"
 
 .if !empty(PKG_OPTIONS:Mdebug)
-CONFIGURE_ARGS.enable+=	asserts verbose-mode
+CONFIGURE_ARGS+=	--enable-asserts
+CONFIGURE_ARGS+=	--enable-verbose-mode
 .else
-CONFIGURE_ARGS.disable+= asserts verbose-mode
+CONFIGURE_ARGS+=	--disable-asserts
+CONFIGURE_ARGS+=	--disable-verbose-mode
 .endif
 
 .if !empty(PKG_OPTIONS:Mkqueue)
-CONFIGURE_ARGS.enable+= 	kqueue
+CONFIGURE_ARGS+= 	--enable-kqueue
 .else
-CONFIGURE_ARGS.disable+=	kqueue
+CONFIGURE_ARGS+=	--disable-kqueue
 .endif
 
 .if !empty(PKG_OPTIONS:Mx11)
-CONFIGURE_ARGS.with+=	x
+CONFIGURE_ARGS+=	--with-x
 .  include "../../x11/libX11/buildlink3.mk"
 BUILDLINK_DEPMETHOD.libXt=	build
 .  include "../../x11/libXt/buildlink3.mk"
 .else
-CONFIGURE_ARGS.without=	x
+CONFIGURE_ARGS=		--without-x
 .endif
 
 .if !empty(PKG_OPTIONS:Mlaunchd)
 MESSAGE_SRC+=			MESSAGE.launchd
 PLIST.launchd=			yes
-CONFIGURE_ARGS.enable+=		launchd
-CONFIGURE_ARGS.with+=		launchd-agent-dir=${PREFIX}/Library/LaunchAgents
+CONFIGURE_ARGS+=		--enable-launchd
+CONFIGURE_ARGS+=		--with-launchd-agent-dir=${PREFIX}/Library/LaunchAgents
+.else
+CONFIGURE_ARGS+=		--disable-launchd
+.endif
+
+.if !empty(PKG_OPTIONS:Menable_in_rcvar)
+DBUS_RCVAR=		dbus_enable
 .else
-CONFIGURE_ARGS.disable+=	launchd
+DBUS_RCVAR=		dbus
 .endif
+
+FILES_SUBST+=		DBUS_RCVAR=${DBUS_RCVAR}
